Codeinwp/neve
# Feat/palette logo

### Summary
The Logo uses a new React component that supports multiple states for light and dark variants.
It defaults and uses the custom logo that was previously set as a default for both states. If palette switcher is not used the behavior should remain the same as before.

While the palette switcher is active you now have the ability to set different variants for light and dark modes.
Since the Palette Switcher uses JS to manually override the browser preferences I could not leverage the `<picture>` tag `media` properties to only switch based on browser preferences so JS is required for the logo switch.

The component uses a listener that detects when the Palette Switcher is added to the Header so that it enables additional features.

The JS from the Logo template is required so that it uses the correct settings from Customizer as well as reacting to the changes from the Palette Switcher using the selective refresh.

### Will affect visual aspect of the product
NO

### Screenshots
Logo when palette switcher is not active:
![image](https://user-images.githubusercontent.com/23024731/126143513-a21bfa0c-0bca-4686-b819-0829514a1fbb.png)

Logo with palette switcher active (2 variants with different and same logo toggle)
![image](https://user-images.githubusercontent.com/23024731/126143763-8a3139de-5c6c-485a-82c5-0b73315cc8e4.png)

![image](https://user-images.githubusercontent.com/23024731/126143907-46f98459-78fa-4362-8080-a7377e434ad8.png)


### Test instructions
1. Test that the Logo behaves the same when Palett Switcher is not active.
2. Add the Palette Switcher to the Header
3. Check that it still behaves the same and that the `Use one logo for both modes` toggle is present on the Logo component.
4. Toggle `Use one logo for both modes` OFF, check that the Logo changes when toggling Light/Dark with the Palette Switcher 

Closes Codeinwp/neve-pro-addon#1067.
